models/DataObject/ClassDefinition/Data/Input.php

```diff
@@ -327,7 +327,20 @@ public function getQueryColumnType()
     public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
     {
         if (!$omitMandatoryCheck && $this->getRegex() && strlen($data) > 0) {
-            if (!preg_match('#' . $this->getRegex() . '#' . implode('', $this->getRegexFlags()), $data)) {
+            $throwException = false;
+            if(in_array('g', $this->getRegexFlags())){
+                $flags = str_replace('g', '', implode('', $this->getRegexFlags()));
+                if (!preg_match_all('#' . $this->getRegex() . '#' . $flags, $data)) {
+                    $throwException = true;
+                }
+            }
+            else{
+                if (!preg_match('#' . $this->getRegex() . '#' . implode('', $this->getRegexFlags()), $data)) {
+                    $throwException = true;
+                }
+            }
+
+            if($throwException) {
```

Comment:
This should be enough, right?

```suggestion
            $flags = str_replace('g', '', implode('', $this->getRegexFlags()));
            if (!preg_match_all('#' . $this->getRegex() . '#' . $flags, $data)) {
```

Reply:
@jdreesen, thanks for your suggestion. 

I think that will cause errors since `preg_match_all` will always act like there is a "g" flag, and since the class editor lets you choose if you want to have the global flag, we shouldn't override that decision.

Or did I miss something?

Reply:
Ooh, I didn't look carefully enough. Forget it, your code is right! Sorry for the noise.

Reply:
@jdreesen no problem 🥇